server/app/auth/oidc/applicant/IdcsProvider.java

```diff
@@ -1,59 +1,102 @@
 package auth.oidc.applicant;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-
 import auth.ProfileFactory;
+import auth.oidc.OidcProvider;
+import com.google.common.collect.ImmutableList;
 import com.google.inject.Inject;
 import com.typesafe.config.Config;
+import java.util.Optional;
 import javax.inject.Provider;
-import org.pac4j.core.http.callback.PathParameterCallbackUrlResolver;
+import org.pac4j.core.profile.creator.ProfileCreator;
 import org.pac4j.oidc.client.OidcClient;
 import org.pac4j.oidc.config.OidcConfiguration;
 import repository.UserRepository;
 
-public class IdcsProvider implements Provider<OidcClient> {
+/** This class customized the OIDC provider to a specific provider, allowing overrides to be set. */
+public class IdcsProvider extends OidcProvider {
```

Comment:
Not blocking: Consider marking final per the above rationale in IdcsProfileAdapter.